.github/workflows/smoke-test.yml

```diff
@@ -0,0 +1,33 @@
+# This workflow will run daily at 06:45.
+# It will install Python dependencies and run tests with a variety of Python versions.
+
+name: Unit test smoke test
+
+on:
+  schedule:
+    - cron: 45 6 * * *
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.9', '3.10', '3.11']
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        python -m pip install --upgrade pip
```

Comment:
This will need the MPI install line, right? 
`sudo apt install -y libopenmpi-dev libhdf5-mpi-dev`